### DISCUSSION TITLE
Max Download speed

### DISCUSSION THREAD

**Discussion by Ockingshay** [transmission/transmission#3194] [Support] [answered]
I recently got gigabit fibre internet but have been using Transmission for years. Max i can get is around 75MB/s where as something like rtorrent hits a stable 110. Also at those speeds transmission becomes incredibly slow/unresponsive, so i simply wait for the download to finish to access the webgui again.

All hardware limitations aside, has anyone managed to get gigabit speeds out of 3.0 or is there a theoretical maximum due to the software?

**Comment by ile6695** [comment]
There are no intentional limitations for download speeds. Transmission 3.0 runs mainly on single thread, which makes it much easier to debug and concurrency-related issues are really rare. One drawback is that when even disk IO is done synchronously, writing data to disk blocks other things, for example UI.

I'm also interested in what could be the maximum, but the answer seems to be: "It depends on hardware". Disk IO can be improved with a ramdisk and that will increase transfer speed.

**Comment by ckerr** [comment, marked as answer]
There is a loooong-standing bug report to make the disk io nonblocking, but it will take a pretty large amount of work and it's already been much too long since there was a new Transmission release, so the current plan is: 4.0.0-beta.1 Real Soon Now, and then the two Big Features for 5.0 will be nonblocking IO and BitTorrent 2.0 support.
